### ISSUE TITLE
StructuredDataClassifier reduces accuracy in learning multiclass classification

### ISSUE THREAD

**Issue by Itsuki-Hamano123** [keras-team/autokeras#1179]
### Bug Description
I tried some multi-class classifications but it didn't work.
loss decreases, auc increases, but accuracy starts from 0.0000e+0 or decreases toward 0.0000e+0.

Even Learning the scikit-learn wine data doesn't seem to work.

### code
`load_data = load_wine(return_X_y =True)

clf = ak.StructuredDataClassifier(max_trials=MAX_TRIALS, seed=7, metrics=["accuracy","AUC"]) 
clf.fit(x=load_data[0], y=load_data[1], epochs=100,  
        #callbacks=[EarlyStopping(monitor='val_loss',patience=2)], 
        validation_split=0.2, 
        verbose=1, 
        )`

### Log during learning
`Starting new trial
Epoch 1/100
5/5 [==============================] - 0s 67ms/step - loss: 114.5369 - accuracy: 0.5188 - auc: 0.5828 - val_loss: 202.5827 - val_accuracy: 0.2037 - val_auc: 0.2500
Epoch 2/100
5/5 [==============================] - 0s 21ms/step - loss: 83.6240 - accuracy: 0.4354 - auc: 0.5828 - val_loss: 168.0761 - val_accuracy: 0.0000e+00 - val_auc: 0.2500
Epoch 3/100
5/5 [==============================] - 0s 15ms/step - loss: 58.2870 - accuracy: 0.4042 - auc: 0.5828 - val_loss: 134.2580 - val_accuracy: 0.0000e+00 - val_auc: 0.2500
Epoch 4/100
5/5 [==============================] - 0s 16ms/step - loss: 33.8346 - accuracy: 0.2438 - auc: 0.5823 - val_loss: 101.0767 - val_accuracy: 0.0000e+00 - val_auc: 0.0000e+00
Epoch 5/100
5/5 [==============================] - 0s 20ms/step - loss: 18.8804 - accuracy: 0.1729 - auc: 0.6094 - val_loss: 89.2839 - val_accuracy: 0.0000e+00 - val_auc: 0.2500
Epoch 6/100
5/5 [==============================] - 0s 15ms/step - loss: 19.1118 - accuracy: 0.2625 - auc: 0.5537 - val_loss: 73.9590 - val_accuracy: 0.0000e+00 - val_auc: 0.2500
Epoch 7/100
5/5 [==============================] - 0s 16ms/step - loss: 14.3848 - accuracy: 0.2375 - auc: 0.5896 - val_loss: 54.5525 - val_accuracy: 0.0000e+00 - val_auc: 0.1528
Epoch 8/100
5/5 [==============================] - 0s 21ms/step - loss: 8.1300 - accuracy: 0.1167 - auc: 0.8059 - val_loss: 40.0003 - val_accuracy: 0.0000e+00 - val_auc: 0.2361
Epoch 9/100
5/5 [==============================] - 0s 15ms/step - loss: 6.0136 - accuracy: 0.0021 - auc: 0.7110 - val_loss: 22.5424 - val_accuracy: 0.0000e+00 - val_auc: 0.0000e+00
Epoch 10/100
5/5 [==============================] - 0s 15ms/step - loss: 2.3730 - accuracy: 0.0021 - auc: 0.7853 - val_loss: 6.1422 - val_accuracy: 0.0000e+00 - val_auc: 0.0000e+00
Epoch 11/100
5/5 [==============================] - 0s 23ms/step - loss: 1.0961 - accuracy: 0.0146 - auc: 0.8628 - val_loss: 0.0368 - val_accuracy: 0.0000e+00 - val_auc: 1.0000
Epoch 12/100
5/5 [==============================] - 0s 31ms/step - loss: 4.2134 - accuracy: 0.0000e+00 - auc: 0.4023 - val_loss: 9.9242 - val_accuracy: 0.0000e+00 - val_auc: 0.0000e+00
Epoch 13/100
5/5 [==============================] - 0s 19ms/step - loss: 3.3954 - accuracy: 0.0479 - auc: 0.7872 - val_loss: 14.7637 - val_accuracy: 0.0000e+00 - val_auc: 0.0000e+00
Epoch 14/100
5/5 [==============================] - 0s 23ms/step - loss: 2.9144 - accuracy: 0.0167 - auc: 0.8290 - val_loss: 12.6576 - val_accuracy: 0.0000e+00 - val_auc: 0.0417
Epoch 15/100
5/5 [==============================] - 0s 17ms/step - loss: 1.6893 - accuracy: 0.0021 - auc: 0.8181 - val_loss: 4.2634 - val_accuracy: 0.0000e+00 - val_auc: 0.1304
Epoch 16/100
5/5 [==============================] - 0s 24ms/step - loss: 0.8308 - accuracy: 0.0083 - auc: 0.8808 - val_loss: 0.1753 - val_accuracy: 0.0000e+00 - val_auc: 1.0000
Epoch 17/100
5/5 [==============================] - 0s 16ms/step - loss: 2.0872 - accuracy: 0.0000e+00 - auc: 0.6184 - val_loss: 4.6047 - val_accuracy: 0.0000e+00 - val_auc: 0.0000e+00
Epoch 18/100
5/5 [==============================] - 0s 22ms/step - loss: 1.6764 - accuracy: 0.0375 - auc: 0.7967 - val_loss: 5.0508 - val_accuracy: 0.0000e+00 - val_auc: 0.0015
Epoch 19/100
5/5 [==============================] - 0s 18ms/step - loss: 0.8066 - accuracy: 0.0146 - auc: 0.8978 - val_loss: 0.9862 - val_accuracy: 0.0000e+00 - val_auc: 0.7014
Epoch 20/100
5/5 [==============================] - 0s 20ms/step - loss: 1.3875 - accuracy: 0.0000e+00 - auc: 0.7424 - val_loss: 1.7443 - val_accuracy: 0.0000e+00 - val_auc: 0.4012
Epoch 21/100
5/5 [==============================] - 0s 25ms/step - loss: 1.0849 - accuracy: 0.0021 - auc: 0.8363 - val_loss: 1.9402 - val_accuracy: 0.0000e+00 - val_auc: 0.3040
Trial complete
Trial summary
|-Trial ID: fbf07580143557b73f310af82ed841d6
|-Score: 0.20370370149612427
|-Best step: 0
Hyperparameters:
|-classification_head_1/dropout_rate: 0
|-optimizer: adam
|-structured_data_block_1/dense_block_1/dropout_rate: 0
|-structured_data_block_1/dense_block_1/num_layers: 2
|-structured_data_block_1/dense_block_1/units_0: 32
|-structured_data_block_1/dense_block_1/units_1: 32
|-structured_data_block_1/dense_block_1/use_batchnorm: False
Starting new trial
Epoch 1/100
5/5 [==============================] - 0s 78ms/step - loss: 1.6491 - accuracy: 0.0000e+00 - auc: 0.3909 - val_loss: 1.5675 - val_accuracy: 0.0000e+00 - val_auc: 0.5000
Epoch 2/100
5/5 [==============================] - 0s 16ms/step - loss: 1.5111 - accuracy: 0.0000e+00 - auc: 0.4947 - val_loss: 0.3868 - val_accuracy: 0.0000e+00 - val_auc: 1.0000
Epoch 3/100
5/5 [==============================] - 0s 16ms/step - loss: 1.2995 - accuracy: 0.0000e+00 - auc: 0.5253 - val_loss: 0.4289 - val_accuracy: 0.0000e+00 - val_auc: 1.0000
Epoch 4/100
5/5 [==============================] - 0s 22ms/step - loss: 1.2213 - accuracy: 0.0000e+00 - auc: 0.5636 - val_loss: 3.0522 - val_accuracy: 0.0000e+00 - val_auc: 0.4823
Epoch 5/100
5/5 [==============================] - 0s 16ms/step - loss: 1.2656 - accuracy: 0.0000e+00 - auc: 0.5293 - val_loss: 4.7503 - val_accuracy: 0.0000e+00 - val_auc: 0.3040
Epoch 6/100
5/5 [==============================] - 0s 16ms/step - loss: 1.2414 - accuracy: 0.0000e+00 - auc: 0.5585 - val_loss: 5.6147 - val_accuracy: 0.0000e+00 - val_auc: 0.1427
Epoch 7/100
5/5 [==============================] - 0s 22ms/step - loss: 1.2245 - accuracy: 0.0000e+00 - auc: 0.5738 - val_loss: 4.4946 - val_accuracy: 0.0000e+00 - val_auc: 0.1304
Epoch 8/100
5/5 [==============================] - 0s 17ms/step - loss: 1.1390 - accuracy: 0.0000e+00 - auc: 0.6079 - val_loss: 3.1674 - val_accuracy: 0.0000e+00 - val_auc: 0.1142
Epoch 9/100
5/5 [==============================] - 0s 18ms/step - loss: 1.0708 - accuracy: 0.0000e+00 - auc: 0.6643 - val_loss: 2.3327 - val_accuracy: 0.0000e+00 - val_auc: 0.1211
Epoch 10/100
5/5 [==============================] - 0s 22ms/step - loss: 1.1398 - accuracy: 0.0000e+00 - auc: 0.6401 - val_loss: 2.2510 - val_accuracy: 0.0000e+00 - val_auc: 0.0170
Epoch 11/100
5/5 [==============================] - 0s 18ms/step - loss: 1.1586 - accuracy: 0.0000e+00 - auc: 0.5927 - val_loss: 2.3412 - val_accuracy: 0.0000e+00 - val_auc: 0.0031
Epoch 12/100
5/5 [==============================] - 0s 28ms/step - loss: 1.1118 - accuracy: 0.0000e+00 - auc: 0.6501 - val_loss: 2.7744 - val_accuracy: 0.0000e+00 - val_auc: 0.0208
Trial complete
Trial summary
|-Trial ID: 3fde209452594df440c7e6c76a977d41
|-Score: 0.0
|-Best step: 0
Hyperparameters:
|-classification_head_1/dropout_rate: 0
|-optimizer: adam
|-structured_data_block_1/dense_block_1/dropout_rate: 0.25
|-structured_data_block_1/dense_block_1/num_layers: 2
|-structured_data_block_1/dense_block_1/units_0: 256
|-structured_data_block_1/dense_block_1/units_1: 128
|-structured_data_block_1/dense_block_1/use_batchnorm: True`

**Comment by haifeng-jin** [comment]
Would you provide your dataset? I can take a look. Thank you!

**Comment by Itsuki-Hamano123** [comment]
Thank you for your reply.
Wine data is provided scikit-learn datasets.
[https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_wine.html#sklearn.datasets.load_wine](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_wine.html#sklearn.datasets.load_wine)
It is table data for three-class classification, and all features are numerical values.

**Comment by haifeng-jin** [comment]
This problem is fixed in the latest release. 1.0.5
